### PR TITLE
enh(database): add option to use unattended with mysql 

### PR DESCRIPTION
**Pull request by paul-oureib**
## Description

add an option to install centreon with mysql as a dependency instead of mariadb 

**Fixes** # MON-23401

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

## Summary by CodeRabbit

- **New Features**
    - Added support for specifying the default database system (MariaDB or MySQL) during setup.
    - Introduced a new option to select the preferred database management system.
    - Enhanced security measures for database setup.
- **Improvements**
    - Streamlined the installation process to accommodate the choice between MariaDB and MySQL.
    - Improved service handling based on the database system selected.

### CODE REVIEW

File: centreon/unattended.sh [hunk 1]
```diff
@@ -602,37 +673,50 @@ function set_runtime_selinux_mode() {
...
+		log "INFO" "Executing SQL requests for $database_system"
+		mysql -u root <<-EOF
+			ALTER USER 'root'@'localhost' IDENTIFIED WITH 'mysql_native_password' BY '${db_root_password}';
+			DELETE FROM mysql.user WHERE User='';
+			DELETE FROM mysql.user WHERE User='root' AND Host NOT IN ('localhost', '127.0.0.1', '::1');
+			DROP DATABASE IF EXISTS test;
+			DELETE FROM mysql.db WHERE Db='test' OR Db='test\\_%';
+			FLUSH PRIVILEGES;
+		EOF
+		systemctl restart mysqld
```
Comment (kduret) [inline centreon/unattended.sh:709]:
```suggestion
```

File: centreon/unattended.sh [hunk 2]
```diff
@@ -602,37 +673,50 @@ function set_runtime_selinux_mode() {
...
-	log "INFO" "Executing SQL requests"
-	mysql -u root <<-EOF
-		UPDATE mysql.global_priv SET priv=json_set(priv, '$.plugin', 'mysql_native_password', '$.authentication_string', PASSWORD('$mariadb_root_password')) WHERE User='root';
-		DELETE FROM mysql.global_priv WHERE User='root' AND Host NOT IN ('localhost', '127.0.0.1', '::1');
-		DELETE FROM mysql.global_priv WHERE User='';
-		DROP DATABASE IF EXISTS test;
-		DELETE FROM mysql.db WHERE Db='test' OR Db='test\\_%';
-		FLUSH PRIVILEGES;
-	EOF
+	log "INFO" "Restarting $database_system service first"
```
Comment (kduret) [inline centreon/unattended.sh:686]:
```suggestion
	log "INFO" "Restarting $database_system service first"
	systemctl daemon-reload
```

File: centreon/unattended.sh [hunk 3]
```diff
@@ -764,7 +855,9 @@ function play_install_wizard() {
...
 	install_wizard_post ${sessionID} "process_step5.php" "admin_password=${centreon_admin_password}&confirm_password=${centreon_admin_password}&firstname=${centreon_admin_firstname}&lastname=${centreon_admin_lastname}&email=${centreon_admin_email}"
-	install_wizard_post ${sessionID} "process_step6.php" "address=&port=3306&root_user=root&root_password=${mariadb_root_password}&db_configuration=centreon&db_storage=centreon_storage&db_user=centreon&db_password=${mariadb_centreon_password}&db_password_confirm=${mariadb_centreon_password}"
+	install_wizard_post ${sessionID} "process_step6.php" "address=&port=3306&root_user=root&root_password=${db_root_password}&db_configuration=centreon&db_storage=centreon_storage&db_user=centreon&db_password=${db_centreon_password}&db_password_confirm=${db_centreon_password}"
+	sed -i "s/\`macros_filter\` TEXT DEFAULT '',/\`macros_filter\` TEXT DEFAULT \(''\),/" /usr/share/centreon/www/install/createTables.sql
+	systemctl restart mysqld
```
Comment (kduret) [inline centreon/unattended.sh]:
```suggestion
```

**Review by kduret — APPROVED** [review]

File: centreon/unattended.sh [hunk 4]
```diff
@@ -31,7 +33,7 @@ function genpasswd() {
 	PWD_LOWER=$(cat /dev/urandom | tr -dc 'a-z' | head -c4)
 	PWD_UPPER=$(cat /dev/urandom | tr -dc 'A-Z' | head -c4)
 	PWD_DIGIT=$(cat /dev/urandom | tr -dc '0-9' | head -c4)
-	PWD_SPECIAL=$(cat /dev/urandom | tr -dc '\!\@\$\*\?' | head -c4)
+	PWD_SPECIAL=$(cat /dev/urandom | tr -dc '\!\@\*\?' | head -c4)
```
[the remark this answers is not in the document]
Comment (tuntoja) [reply centreon/unattended.sh:36]: this is intended to reduce password complexity while still being compliant to password policy

File: centreon/unattended.sh [hunk 5]
```diff
@@ -1160,11 +1228,17 @@ function install_poller() {
 # update Centreon packages
 #
 function update_centreon_packages() {
+	if [[ $dbms == "MariaDB" ]]; then
```
Comment (tuntoja) [inline centreon/unattended.sh:1231]: why is the set of package to be updated different depending on the dbms used ?

File: centreon/unattended.sh [hunk 6]
```diff
@@ -1131,8 +1200,8 @@ function install_central() {
 
 	log "INFO" "PHP date.timezone set to [$timezone]"
 
-	secure_mariadb_setup
-}
+	secure_dbms_setup
+}
```
Comment (kduret) [inline centreon/unattended.sh:1204]: do not add trailing spaces :)

**Review by kduret — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

File: centreon/unattended.sh [hunk 7]
```diff
@@ -351,30 +371,44 @@ function set_centreon_repos() {
...
-		else
-			log "INFO" "Successfully installed MariaDB repository"
-		fi
+	if [[ "${detected_os_release}" =~ debian-release-.* ]]; then
+		curl -LsS https://r.mariadb.com/downloads/mariadb_repo_setup | bash -s -- --os-type=debian --os-version="$detected_os_version" --mariadb-server-version="$detected_mariadb_version"
+	else
+		curl -LsS https://r.mariadb.com/downloads/mariadb_repo_setup | bash -s -- --mariadb-server-version="$detected_mariadb_version"
+	fi
+	if [ $? -ne 0 ]; then
+		error_and_exit "Could not install the repository"
```
Comment (tuntoja) [inline centreon/unattended.sh:380]: add dbms name of the repository

File: centreon/unattended.sh [hunk 8]
```diff
@@ -1256,17 +1324,17 @@ esac
...
 	if [ "$wizard_autoplay" == "true" ]; then
-		# Set from ENV or random MariaDB centreon password
-		mariadb_centreon_password=${ENV_MARIADB_CENTREON_PASSWD:-"$(genpasswd "MariaDB user: centreon")"}
+		# Set from ENV or random Database centreon password
+		db_centreon_password=${ENV_DB_CENTREON_PASSWD:-"$(genpasswd "Database user: centreon")"}
 		# Generate random password if Centreon admin password is empty
 		if [ -z "${centreon_admin_password}" ]; then
 			centreon_admin_password=${ENV_CENTREON_ADMIN_PASSWD:-"$(genpasswd "Centreon user: admin")"}
 		else
 			test_password_policy
```
Comment (tuntoja) [inline centreon/unattended.sh:1336]: should this one be re-indented as well ?

File: centreon/unattended.sh [hunk 9]
```diff
@@ -1256,17 +1324,17 @@ esac
...
+	db_root_password=${ENV_DB_ROOT_PASSWD:-"$(genpasswd "Database user: root")"}
 
 	if [ "$wizard_autoplay" == "true" ]; then
-		# Set from ENV or random MariaDB centreon password
-		mariadb_centreon_password=${ENV_MARIADB_CENTREON_PASSWD:-"$(genpasswd "MariaDB user: centreon")"}
+		# Set from ENV or random Database centreon password
+		db_centreon_password=${ENV_DB_CENTREON_PASSWD:-"$(genpasswd "Database user: centreon")"}
 		# Generate random password if Centreon admin password is empty
 		if [ -z "${centreon_admin_password}" ]; then
 			centreon_admin_password=${ENV_CENTREON_ADMIN_PASSWD:-"$(genpasswd "Centreon user: admin")"}
```
Comment (tuntoja) [inline centreon/unattended.sh:1334]: should this one be re-indented as well ?

**Review by tuntoja — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**